src/qiboml/models/ansatze.py

```diff
@@ -12,21 +12,74 @@ def HardwareEfficient(
     nqubits: int,
     qubits: Optional[tuple[int]] = None,
     nlayers: int = 1,
+    single_block: Optional[Circuit] = None,
+    entangling_block: Optional[Circuit] = None,
+    entangling_gate: str = "CNOT",
+    architecture: str = "diagonal",
+    closed_boundary: bool = True,
+    **kwargs,
 ) -> Circuit:
+    """
+    Create a Hardware-efficient ansatz with custom single-qubit and entangling blocks.
+
+    Args:
+        nqubits (int): Number of qubits in the ansatz.
+        qubits (tuple[int], optional): Qubit indices to apply the ansatz to. If ``None``, the ansatz is applied to all qubits
+            from ``0`` to ``nqubits-1``. Defaults to ``None``.
+        nlayers (int, optional): Number of layers (single-qubit + entangling per layer). Defaults to 1.
+        single_block (Circuit, optional): 1-qubit Circuit applied to each qubit. Defaults to a block with :class:`qibo.gates.RY` and :class:`qibo.gates.RZ` gates.
+        entangling_block (Circuit, optional): full n-qubit entangling circuit. Defaults to ``None``.
```

Comment:
```suggestion
        entangling_block (Circuit, optional): :math:`n`-qubit entangling circuit. Defaults to ``None``.
```
What happens if `None`?